[Tensor Parallel] Add a new Colwise Parallel style when Pairwise cannot directly used

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100137

Some use cases, users cannot directly `PairwiseParallelStyle` and they might need to specify colwise and rowwise separately.

Commit:
[Tensor Parallel] Add a new Colwise Parallel style when Pairwise cannot directly used

[ghstack-poisoned]

Commit:
Update on "[Tensor Parallel] Add a new Colwise Parallel style when Pairwise cannot directly used"




[ghstack-poisoned]

lessw2020 commented on torch/distributed/tensor/parallel/api.py:
```diff
@@ -271,7 +272,7 @@ def _parallelize_linear(
             output_fn=parallel_style._prepare_output,  # type: ignore[arg-type, misc] # pyre-ignore[6]
         )
-    elif isinstance(parallel_style, ColwiseParallel):
+    elif isinstance(parallel_style, ColwiseParallel) or isinstance(parallel_style, ColwiseParallelNoReshard):
```
this can be simplified to:
elif isinstance(parallel_style, (ColwiseParallel, ColwiseParallelNoReshard)):

Commit:
Update on "[Tensor Parallel] Add a new Colwise Parallel style when Pairwise cannot directly used"


Some use cases, users cannot directly `PairwiseParallelStyle` and they might need to specify colwise and rowwise separately.

[ghstack-poisoned]

wz337 reviewed (approved):
LGTM

Commit:
Update on "[Tensor Parallel] Add a new Colwise Parallel style when Pairwise cannot directly used"


Some use cases, users cannot directly `PairwiseParallelStyle` and they might need to specify colwise and rowwise separately.

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge
